dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/util/AnnotationUtils.java

```diff
@@ -39,13 +53,186 @@
  */
 public class AnnotationUtils {
 
+    public static String resolveInterfaceName(Service service, Class<?> defaultInterfaceClass)
+            throws IllegalStateException {
+
+        String interfaceName;
+        if (StringUtils.hasText(service.interfaceName())) {
+            interfaceName = service.interfaceName();
+        } else if (!void.class.equals(service.interfaceClass())) {
+            interfaceName = service.interfaceClass().getName();
+        } else if (defaultInterfaceClass.isInterface()) {
+            interfaceName = defaultInterfaceClass.getName();
+        } else {
+            throw new IllegalStateException(
+                    "The @Service undefined interfaceClass or interfaceName, and the type "
+                            + defaultInterfaceClass.getName() + " is not a interface.");
+        }
+
+        return interfaceName;
+
+    }
+
+    public static String resolveInterfaceName(Reference reference, Class<?> defaultInterfaceClass)
+            throws IllegalStateException {
+
+        String interfaceName;
+        if (!"".equals(reference.interfaceName())) {
+            interfaceName = reference.interfaceName();
+        } else if (!void.class.equals(reference.interfaceClass())) {
+            interfaceName = reference.interfaceClass().getName();
+        } else if (defaultInterfaceClass.isInterface()) {
+            interfaceName = defaultInterfaceClass.getName();
+        } else {
+            throw new IllegalStateException(
+                    "The @Reference undefined interfaceClass or interfaceName, and the type "
+                            + defaultInterfaceClass.getName() + " is not a interface.");
+        }
+
+        return interfaceName;
+
+    }
+
+
+    // Cloned from https://github.com/alibaba/spring-context-support/blob/1.0.2/src/main/java/com/alibaba/spring/util/AnnotationUtils.java
+
     /**
-     * Get {@link Annotation} attributes
+     * Is specified {@link Annotation} present on {@link Method}'s declaring class or parameters or itself.
      *
-     * @param annotation
-     * @param propertyResolver
-     * @param ignoreDefaultValue
+     * @param method          {@link Method}
+     * @param annotationClass {@link Annotation} type
+     * @param <A>             {@link Annotation} type
+     * @return If present , return <code>true</code> , or <code>false</code>
+     * @since 2.6.6
+     */
+    public static <A extends Annotation> boolean isPresent(Method method, Class<A> annotationClass) {
+
+        Map<ElementType, List<A>> annotationsMap = findAnnotations(method, annotationClass);
+
+        return !annotationsMap.isEmpty();
+
+    }
```

Comment:
I found is is no usage. There is no class to invoke this method.

Reply:
ignore .  there are some code that is introduced by org.springframework.core.annotation.AnnotationUtils.